import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/AMPStaticProcessor.java

```diff
@@ -452,145 +454,152 @@ private void processPercentage(List<JsonBean> percentageObject, String percentag
+			@Override
+			public ActionResult get() {
+				ActionResult result;
 
-		} catch (RestClientException e) {
-			log.error("Error importing activity " + e);
-			e.printStackTrace();
-			if (e.getClass().equals(HttpServerErrorException.class)) {
-				HttpServerErrorException ex = (HttpServerErrorException) e;
-				JsonBean resultPost = JsonBean.getJsonBeanFromString(ex.getResponseBodyAsString());
-				if (resultPost != null) {
-					Object errorNode = resultPost.get("error");
-					Map<?, ?> activity = (Map<?, ?>) resultPost.get("activity");
-					Object projectTitle = (activity != null && activity.get("project_title") != null)
-							? activity.get("project_title")
-							: "";
-					result = new ActionResult("N/A", "ERROR", "ERROR",
-							"REST Exception:" + projectTitle + " " + errorNode);
-				} else {
-					result = new ActionResult("N/A", "ERROR", "ERROR", "REST Exception:" + e.getMessage());
+				String operation = Constants.AMP_INSERT_OPERATION;
+				String url = Constants.AMP_ACTIVITY_ENDPOINT;
+				if (mappedProject.getProject().get(Constants.AMP_INTERNAL_ID) != null) {
+					operation = Constants.AMP_UPDATE_OPERATION;
+					url += "/" + mappedProject.getProject().getString(Constants.AMP_INTERNAL_ID);
 				}
-			} else if (e.getClass().equals(HttpClientErrorException.class)) {
-				HttpClientErrorException ex = (HttpClientErrorException) e;
-				JsonBean resultPost = JsonBean.getJsonBeanFromString(ex.getResponseBodyAsString());
-				if (resultPost != null) {
+				try {
+					JsonBean resultPost = restTemplate.postForObject(baseURL + url, mappedProject.getProject(), JsonBean.class);
+
 					Object errorNode = resultPost.get("error");
-					Map<?, ?> activity = (Map<?, ?>) resultPost.get("activity");
-					Object projectTitle = (activity != null && activity.get("project_title") != null)
-							? activity.get("project_title")
-							: "";
-					result = new ActionResult("N/A", "ERROR", "ERROR",
-							"REST Exception:" + projectTitle + " " + errorNode);
-				} else {
-					result = new ActionResult("N/A", "ERROR", "ERROR", "REST Exception:" + e.getMessage());
+
+					if (errorNode == null) {
+						Integer id = (int) resultPost.get("internal_id");
+						String message = "";
+						if (resultPost.get("project_title") instanceof Map) {
+							@SuppressWarnings("unchecked")
+							Map<String, String> titleMultilang = (Map<String, String>) resultPost.get("project_title");
+							message = (String) titleMultilang.entrySet().stream().map(i -> i.getValue())
+									.collect(Collectors.joining(", "));
+						} else {
+							message = resultPost.getString("project_title");
+						}
+						result = new ActionResult(id.toString(), operation, "OK", message,
+								null, null);
+					} else {
+						String error = errorNode.toString();
+						result = new ActionResult("N/A", "REJECT", "ERROR", "Error: " + error);
+					}
+				} catch (RestClientException e) {
+					log.error("Error importing activity " + e);
+					e.printStackTrace();
```

Comment:
Since we are logging the error using the logger, please remove e.printStackTrace();